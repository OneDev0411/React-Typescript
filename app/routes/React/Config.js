// Config.js
import React from 'react'
import { Route, IndexRoute, Redirect } from 'react-router'

// Main component
import App from '../../components/App'

// Pages
import Landing from '../../components/Pages/Landing'
import Recents from '../../components/Pages/Dashboard/Recents'
import Mls from '../../components/Pages/Dashboard/Mls'
import Contacts from '../../components/Pages/Dashboard/Contacts'
import Tasks from '../../components/Pages/Dashboard/Tasks'
import Transactions from '../../components/Pages/Dashboard/Transactions'
import SingleTransaction from '../../components/Pages/Dashboard/Transactions/Single'
import NewTransaction from '../../components/Pages/Dashboard/Transactions/New'
import SignUp from '../../components/Pages/SignUp'
import SignIn from '../../components/Pages/SignIn'
import Verify from '../../components/Pages/Verify'
import Password from '../../components/Pages/Password'
import NoMatch from '../../components/Pages/NoMatch'
import Settings from '../../components/Pages/Account/Settings'
import Notifications from '../../components/Pages/Account/Notifications'
import Recommend from '../../components/Pages/Dashboard/Mls/Listing/Recommend'
import Agents from '../../components/Pages/Dashboard/Mls/Agents'

export default (
  <Route path="/" component={App}>
    <IndexRoute component={Landing}/>
    <Route path="signup" component={SignUp}/>
    <Route path="signin" component={SignIn}/>
    <Route path="account/settings" component={Settings}/>
    <Route path="account/notifications" component={Notifications}/>
    <Redirect from="dashboard" to="dashboard/recents" />
    <Route path="dashboard/recents" component={Recents}/>
    <Route path="dashboard/recents/:room_id" component={Recents}/>
    <Route path="dashboard/mls" component={Mls}/>
    <Route path="dashboard/mls/agents" component={Agents}/>
    <Route path="dashboard/mls/listing/recommend" component={Recommend}/>
    <Route path="dashboard/contacts" component={Contacts}/>
    <Route path="dashboard/contacts/:id" component={Contacts}/>
    <Route path="dashboard/tasks" component={Tasks}/>
    <Route path="dashboard/transactions" component={Transactions}/>
    <Route path="dashboard/transactions/new" component={NewTransaction}/>
    <Route path="dashboard/transactions/:id" component={SingleTransaction}/>
    <Route path="dashboard/transactions/:id/attachments/:id" component={SingleTransaction}/>
    <Route path="verify/:slug" component={Verify}/>
    <Route path="password/:slug" component={Password}/>
    <Route path="*" component={NoMatch}/>
  </Route>
)