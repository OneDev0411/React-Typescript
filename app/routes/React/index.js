// index.js
import React from 'react'
import { match, RoutingContext } from 'react-router'
import ReactDOMServer from 'react-dom/server'

// Store
import AppStore from '../../stores/AppStore'

// Config
import routes from './Config'

module.exports = (app, config) => {

  app.get('*',(req, res) => {

    match({ routes, location: req.url }, (error, redirectLocation, renderProps) => {
      
      let reactMarkup = ReactDOMServer.renderToStaticMarkup(<RoutingContext data={AppStore.data} {...renderProps} />)
      
      res.locals.reactMarkup = reactMarkup

      console.log(reactMarkup)

      if (error) {
      
        res.status(500).send(error.message)
      
      } else if (redirectLocation) {
        
        res.redirect(302, redirectLocation.pathname + redirectLocation.search)

      } else if (renderProps) {
        
        // Success!
        res.status(200).render('index.html')
      
      } else {

        res.status(404).render('index.html')
      
      }

    })
  })
  
}