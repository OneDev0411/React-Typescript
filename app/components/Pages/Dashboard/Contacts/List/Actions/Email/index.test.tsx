import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'

// @ts-ignore
// eslint-disable-next-line
import recipient from 'fixtures/contacts/contact-item'

import { Email } from '.'

const contact = recipient as INormalizedContact

describe('ContactItem', () => {
  it('should render', () => {
    const wrapper = shallow(<Email contacts={[contact]} />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
