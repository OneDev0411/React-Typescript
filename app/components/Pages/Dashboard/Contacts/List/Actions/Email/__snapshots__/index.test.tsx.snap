// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactItem should render 1`] = `
<Connect(SendEmailButton)
  recipients={
    Array [
      Object {
        "contact": Object {
          "avatar": null,
          "contactId": "350b2dd0-c270-4a14-a257-87ab1e23a767",
          "data_type": "contact",
          "email": "alireza.mirian@gmail.com",
          "emails": Array [
            "alireza.mirian@gmail.com",
            "snhasani@gmail.com",
            "naser@rechat.com",
            "naser@rechat.com",
          ],
          "name": "Nasser Hassani",
          "users": Array [],
        },
        "email": "alireza.mirian@gmail.com",
        "recipient_type": "Email",
      },
    ]
  }
  style={
    Object {
      "fontSize": "0.875rem",
    }
  }
/>
`;
